janusgraph-server/src/test/java/org/janusgraph/graphdb/server/JanusGraphManagerImplTest.java

```diff
@@ -0,0 +1,136 @@
+import org.javatuples.Pair;
+import org.junit.jupiter.api.Test;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class JanusGraphManagerImplTest {
+
+    private GraphManager getGraphManager() {
+        Settings settings = new Settings();
+        HashMap<String, String> map = new HashMap<>();
+        map.put("graph", "src/test/resources/janusgraph-inmemory.properties");
+        map.put("graph2", "src/test/resources/janusgraph-inmemory.properties");
+        map.put("tinkergraph", "src/test/resources/tinkergraph.properties");
+        settings.graphs = map;
+        return new DefaultGraphManager(settings);
+    }
+
+    private Pair<Server, String> createServer(GraphManager graphManager) throws IOException {
+        String serverName = InProcessServerBuilder.generateName();
+        Server server = InProcessServerBuilder
+            .forName(serverName).directExecutor().addService(new JanusGraphManagerImpl(graphManager)).build().start();
+        return new Pair<>(server, serverName);
+    }
+
+    private Pair<TestingServerClosable, JanusGraphManagerGrpc.JanusGraphManagerBlockingStub> createServerStub(GraphManager graphManager) throws IOException {
+        Pair<Server, String> server = createServer(graphManager);
+        ManagedChannel channel = InProcessChannelBuilder.forName(server.getValue1()).directExecutor().build();
+        JanusGraphManagerGrpc.JanusGraphManagerBlockingStub stub = JanusGraphManagerGrpc.newBlockingStub(channel);
+        TestingServerClosable testingServerClosable = new TestingServerClosable(server.getValue0(), channel);
+        return new Pair<>(testingServerClosable, stub);
+    }
+
+    @Test
+    public void testContextIsReturnedForGivenGraphName() throws IOException {
+        GraphManager graphManager = getGraphManager();
+        Pair<TestingServerClosable, JanusGraphManagerGrpc.JanusGraphManagerBlockingStub> serverStub = createServerStub(graphManager);
+
+        JanusGraphContext test = serverStub.getValue1().getJanusGraphContextByGraphName(
+            GetJanusGraphContextByGraphNameRequest.newBuilder().setGraphName("graph").build());
+
+        assertEquals("graph", test.getGraphName());
+        serverStub.getValue0().close();
+    }
+
+    @Test
+    public void testErrorIsReturnedForGivenGraphNameWhichIsNotAJanusGraph() throws IOException {
+        GraphManager graphManager = getGraphManager();
+        Pair<TestingServerClosable, JanusGraphManagerGrpc.JanusGraphManagerBlockingStub> serverStub = createServerStub(graphManager);
+
+        assertThrows(StatusRuntimeException.class, () ->
+            serverStub.getValue1().getJanusGraphContextByGraphName(
+                GetJanusGraphContextByGraphNameRequest.newBuilder().setGraphName("tinkergraph").build()));
```

Comment:
Nit: These two lines are often repeated so you could move them to a method which then takes the graph name as a parameter and returns the JanusGraphContext.